core/src/mindustry/core/UI.java

```diff
@@ -307,6 +307,15 @@ public void showTextInput(String titleText, String dtext, int textLength, String
                     hide();
                 }).disabled(b -> field.getText().isEmpty());
                 buttons.addButton("$cancel", this::hide);
+                this.keyDown(KeyCode.ENTER, () -> {
```

Comment:
Don't use `this` unless necessary. You use it here 3  times without a purpose.

Reply:
Thank you for the notice. Didn't know this wasn't necessary in Java.
I pushed the changes for further investigation.